src/ranking.jl

```diff
@@ -171,3 +171,178 @@ Missing values are assigned rank `missing`.
+a quantile rank of x means that x% of the elements in `v` are lesser or lesser-equal the 
+given `value`. 
+
+The keyword argument `method` (default `:inc`) can be:
+
+`:inc` - The inverse of the quantile based in def. 7 in Hyndman and Fan (1996). 
+(Excel `PERCENTRANK` and `PERCENTRANK.INC`)
+
+`:exc` - The inverse of the quantile based in def. 6 in Hyndman and Fan (1996). 
+(Excel `PERCENTRANK.EXC`)
+
+`:compete` - Based on the `competerank` of StatsBase 
+(MariaDB `PERCENT_RANK`, dplyr `percent_rank`)
+
+`:tied` - Based in the def. in Roscoe, J. T. (1975) 
+(`mean` argument of Scipy `percentileofscore`)
+
+`:strict` - (`strict` argument of Scipy `percentileofscore`)
+
+`:weak` - (`weak` argument of Scipy `percentileofscore`)
+
+!!! note
+    An `ArgumentError` is thrown if `v` contains `NaN` or `missing` values
+    and if `v` is empty or contains only one element.
+
+# References
+[Percentile Rank on Wikipedia](https://en.wikipedia.org/wiki/Percentile_rank) covers 
+definitions and examples.
+
+Roscoe, J. T. (1975). "[Fundamental Research Statistics for the Behavioral Sciences 
+(2nd ed.)](http://www.bryanburnham.net/wp-content/uploads/2014/07/Fundamental-Statistics-
+for-the-Behavioral-Sciences-v2.0.pdf#page=57)".
+    ISBN 0-03-091934-7.
+
+Hyndman, R.J and Fan, Y. (1996) "[Sample Quantiles in Statistical Packages]
+(https://www.amherst.edu/media/view/129116/original/Sample+Quantiles.pdf)",
+    *The American Statistician*, Vol. 50, No. 4, pp. 361-365
+
+[Quantile on Wikipedia](https://en.m.wikipedia.org/wiki/Quantile) details the different 
+quantile definitions.
+
+# Examples
+```julia
+julia> using StatsBase
+
+julia> v1 = [1, 1, 1, 2, 3, 4, 8, 11, 12, 13];
+
+julia> v2 = [1, 2, 3, 4, 4, 5, 6, 7, 8, 9];
+
+julia> quantilerank(v1, 2), quantilerank(v1, 8)
+(0.3333333333333333, 0.5555555555555556)
+
+# use `Ref` to treat vector `v2` as a scalar during broadcasting.
+julia> quantilerank.(Ref(v2), [4, 8])
+2-element Vector{Float64}:
+ 0.3333333333333333
+ 0.8888888888888888
+ 
+# also works with non-number types
+julia> using Dates
+
+julia> d1 = Date("2021-01-01");
+
+julia> daterange = d1:Day(1):d1+Day(99);
+
+julia> quantilerank(daterange, d1 + Day(20))
+0.20202020202020202
+```
+"""
+function quantilerank(v::AbstractVector, value; method::Symbol=:inc)
+    # checks
+    value isa Number && isnan(value) &&
+        throw(ArgumentError("value cannot be NaN"))
+    any(x -> ismissing(x) || (x isa Number && isnan(x)), v) &&
+        throw(ArgumentError("vector `v` cannot contain missing or NaN entries"))
+
+    n = length(v)
+
+    n == 0 && throw(ArgumentError("vector `v` is empty. Insert a non-empty vector"))
+    n == 1 && throw(ArgumentError("vector `v` has only 1 value. Use a vector with more elements"))
+
+    if method == :inc
+        if value > maximum(v)
+            return 1.0
+        elseif value ≤ minimum(v)
+            return 0.0
+        else
+            !issorted(v) && (v = sort(v))
+            v[1] == value && return 0.0
```

Comment:
here - and in other places the code is incorrect as it assumes 1-based indexing.

Reply:
Great point. I will be aware of this in the next few times and I will use the word "begin". Tks.